src/utils/__tests__/addresses.test.ts

```diff
@@ -0,0 +1,82 @@
+import { checksumAddress, isChecksummedAddress, parsePrefixedAddress, sameAddress } from '../addresses'
+
+describe('Addresses', () => {
+  describe('checksumAddress', () => {
+    it('should checksum addresses', () => {
+      const value = checksumAddress('0x62da87ff2e2216f1858603a3db9313e178da3112')
+      expect(value).toBe('0x62Da87FF2E2216F1858603A3Db9313E178da3112')
+    })
+
+    it('should return checksummed addresses as is', () => {
+      const value = checksumAddress('0x62Da87FF2E2216F1858603A3Db9313E178da3112')
+      expect(value).toBe('0x62Da87FF2E2216F1858603A3Db9313E178da3112')
+    })
+
+    it('should return non-addresses as is', () => {
+      const value = checksumAddress('sdfgsdfg')
+      expect(value).toBe('sdfgsdfg')
+    })
```

Comment:
I'd add another interesting testcase which is an incorrect mixed case address(with wrong checksum) should still return the same incorrect mixed case address